dali/base/dadfs.cpp

```diff
@@ -6116,6 +6122,13 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
                 throw MakeStringException(-1, "The value of @numsubfiles (%d) is not equal to the number of SubFile items (%d)!",numSubfiles, subFileCount);
         }
     }
+    virtual bool isRestrictedAccess() const override
+    {
+        // This ensures restriction applies even if superfile is unrestricted but subfile are.
+        // However, this also means that the superfile will show as "Restricted" in ECL Watch and whenever the user tries to unset the flag
+        // it will appear to reset to Restricted. (TODO)
+        return subfileRestricted || root->getPropBool("@restricted", false);
```

Comment:
Needs changing, inlight of other PR-12940.
How could subfileRestricted be false, but the getPropBool be true here?

Reply:
Yes, it can be and yes they'd probably be able access the subfile directly if that was the case.  Should we make it so that whenever the superfile is marked as restricted, all subfiles are marked as restricted?

Reply:
If a super file is not restricted, but all subs are, then it is fully restricted (from reading any content) anyway.
If super is restricted, but some subs are not, then they can always bypass the super and go to the subs - so don't think restricting the super adds anything, *unless* reading a subfile (looking it up in DFS), involved seeing if was owned by a super that was restricted. Which is kind of similar to trying to delete a subfile that is owned by a super (you get an error).

I would not do anything special for supers for now.